src/com/tapia/foss/frames/Visualizer.java

```diff
@@ -232,7 +238,13 @@ private void jButtonAddActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FI
         }
         else
         {
-            structure.add(Integer.parseInt(input));
+            
+            if (this.tryParsing(input)) {
+                structure.add(Integer.parseInt(input));
+            } else {
+                return;
```

Comment:
Instead of returning, how about we give a message to the user using JOptionPane giving feedback on sending in an integer only.
I dont think return is necessary here, is it?